### PR TITLE
add distributed tracing

### PR DESCRIPTION
**Pull request by ktoso**
### Pull Request Description

adds distributed tracing repo.

Important to run both 5.0 and 5.2 since the project has a bunch of features only available on 5.2

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 4.0 and passes any unit tests
  - 5.0+
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* Apache License, version 2.0
- [x] pass `./project_precommit_check` script run

```
PASS: swift-distributed-tracing, 5.2, 73f2b5, Swift Package
========================================
Action Summary:
     Passed: 1
     Failed: 0
    XFailed: 0
    UPassed: 0
      Total: 1
========================================
Repository Summary:
      Total: 1
========================================
Result: PASS
========================================
--- swift-distributed-tracing checked successfully against Swift 5.2 ---
```

### CODE REVIEW

**Comment by shahmishal** [comment]
Can you also add tag `"tags": "swiftpm"`? example: https://github.com/apple/swift-source-compat-suite/blob/main/projects.json#L201

**Comment by shahmishal** [comment]
@swift-ci test

File: projects.json [hunk 1]
```diff
@@ -2963,6 +2963,37 @@
       }
     ]
   },
+  {
+    "repository": "Git",
+    "url": "https://github.com/apple/swift-distributed-tracing",
+    "path": "swift-distributed-tracing",
+    "branch": "master",
```
Comment (shahmishal) [inline projects.json:2970]: This should be `main` branch.

Comment (shahmishal) [inline projects.json:2970]:
```suggestion
    "branch": "main",
```

Comment (ktoso) [reply projects.json]: Thanks, sorry I missed this!

**Comment by ktoso** [comment]
@swift-ci test

File: projects.json [hunk 2]
```diff
@@ -2963,6 +2963,37 @@
...
+    ],
+    "platforms": [
+      "Darwin",
+      "Linux"
+    ],
+    "actions": [
+      {
+        "action": "BuildSwiftPackage",
+        "configuration": "release",
+        "tags": "sourcekit-disabled swiftpm"
```
Comment (ktoso) [inline projects.json:2990]: Ok I think all PRs I made have the `swiftpm` tag too, seems right @shahmishal?

Thanks

Comment (shahmishal) [reply projects.json:2990]: Yes, that looks good.

**Review by shahmishal — APPROVED** [review]
Thanks!

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by tomerd** [comment]
@ktoso looks like it conflicts now, could you resolve so this can be merged

**Comment by ktoso** [comment]
@swift-ci test

**Comment by ktoso** [comment]
@swift-ci test and merge

**Comment by ktoso** [comment]
@swift-ci test

**Pull request merged**